pkg/loki/loki.go

```diff
@@ -280,8 +280,23 @@ func newDefaultConfig() *Config {
 	return defaultConfig
 }
 
+// RunOpts configures custom behavior for running Loki.
+type RunOpts struct {
+	// CustomConfigEndpointHandlerFn is the handlerFunc to be used by the /config endpoint.
+	// If empty, default handlerFunc will be used.
+	CustomConfigEndpointHandlerFn func(http.ResponseWriter, *http.Request)
```

Comment:
Out of curiosity, why is this required?

Reply:
Without this, other projects (ex: GEL) aren't able to customize the `/config` endpoint. That's a problem for projects that uses a superset of configurations that include Loki configs because their configs that aren't part of Loki aren't shown in the /config endpoint.

Reply:
That's a very good point. It relates to https://github.com/grafana/loki-private/issues/96 though :slightly_smiling_face:

Reply:
Yes, it has some relation to it.